src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2720,7 +2721,30 @@ public static void TestNameWithInvariantCulture()
                 Assert.True(pacific.DisplayName.IndexOf("Pacific", StringComparison.OrdinalIgnoreCase) >= 0, $"'{pacific.DisplayName}' is not the expected display name for Pacific time zone");
 
             }).Dispose();
+        }
+
+        private static bool CanTestWindowsNlsDisplayNames => s_isWindows && RemoteExecutor.IsSupported && WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages().Length > 1;
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalFact(nameof(CanTestWindowsNlsDisplayNames))]
+        public static void TestWindowsNlsDisplayNames()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                var cultures = WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages();
+
+                CultureInfo.CurrentUICulture = cultures[0];
+                TimeZoneInfo.ClearCachedData();
+                TimeZoneInfo tz1 = TimeZoneInfo.FindSystemTimeZoneById(s_strPacific);
+
+                CultureInfo.CurrentUICulture = cultures[1];
+                TimeZoneInfo.ClearCachedData();
+                TimeZoneInfo tz2 = TimeZoneInfo.FindSystemTimeZoneById(s_strPacific);
 
+                Assert.True(tz1.StandardName != tz2.DisplayName, $"The display name '{tz1.DisplayName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
```

Comment:
>tz1.StandardName != tz2.DisplayName,  [](http://example.com/codeflow?start=28&length=37)

Isn't possible you can have 2 languages in the language lists which possibly returning exact resources? something like "en" and "en-US" for instance?

Reply:
That's why I have the `GetInstalledWin32CulturesWithUniqueLanguages` method to do a groupby/first on language